src/exoscale/compute/api/http.clj

```diff
@@ -108,8 +108,10 @@
                                                     :pagesize pagesize))
                (fn [resp]
                  (let [acc (concat acc resp)
-                       all-results-present? (= (:count (meta resp)) (count resp))]
-                   (if (or single-page-only?
+                       meta-count (:count (meta resp))
+                       all-results-present? (= meta-count (count acc))]
+                   (if (or (nil? meta-count)
```

Comment:
This used to recur if count was `0` as it was trying to aggregate results, if it's nil we should never recur.